### PR TITLE
add async communication in Voting, not test has been verified

### PR DESCRIPTION
**Pull request by BenomarAchraf**

### CODE REVIEW

File: auth-service/controllers/authController.js [hunk 1]
```diff
@@ -59,12 +61,19 @@ exports.login = async (req, res) => {
     // Generate and send a JWT token
     const token = jwt.sign({ userId: user._id }, secret, { expiresIn });
 
-    res.json({ message: 'Login successful', token, userId: user._id, email: user.email, isAdmin: user.isAdmin, isEligibleToApply: user.isEligibleToApply,isEligibleToVote: user.isEligibleToVote, hasApplied: user.hasApplied, hasVoted: user.hasVoted });
+    res.json({
```
Comment (OssamaLouati) [inline auth-service/controllers/authController.js:64]: Formatting should be fixed

File: auth-service/controllers/voteController.js [hunk 2]
```diff
@@ -20,6 +20,26 @@ exports.verifyHasVoted = async (req, res) => {
       .json({ message: error + " Server error when vertifyHasVoted" });
   }
 };
+exports.verifyifusercanVote = async (req, res) => {
```
Comment (OssamaLouati) [inline auth-service/controllers/voteController.js]: this controller should be removed

File: paiment-service/controllers/Async/publisher.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
+const amqp = require("amqplib/callback_api");
+
```
Comment (OssamaLouati) [inline paiment-service/controllers/Async/publisher.js:2]: Can't decide weather this is relevant

File: paiment-service/controllers/Async/subscriber.js [hunk 4]
```diff
@@ -0,0 +1,47 @@
+const amqp = require("amqplib/callback_api");
```
Comment (OssamaLouati) [inline paiment-service/controllers/Async/subscriber.js:1]: same thing here

File: paiment-service/controllers/PaymentController.js [hunk 5]
```diff
@@ -1,11 +1,12 @@
+const { response } = require("express");
 const Payment = require("../models/Payment");
 
 exports.createPaymentAccount = async (req, res) => {
```
Comment (OssamaLouati) [inline paiment-service/controllers/PaymentController.js]: this controller should be removed based on achraf

File: paiment-service/controllers/PaymentController.js [hunk 6]
```diff
@@ -19,10 +20,47 @@ exports.createPaymentAccount = async (req, res) => {
...
+
+    await newPayment.save();
+
+    console.log({ message: "Your Payment has been registered successfully" });
+  } catch (error) {
+    console.error(error);
+  }
+};
+
+exports.PaymentAsync = async (msg) => {
```
Comment (OssamaLouati) [inline paiment-service/controllers/PaymentController.js]: incompleted controller, should be tested and finished

File: paiment-service/controllers/PaymentController.js [hunk 7]
```diff
@@ -19,10 +20,47 @@ exports.createPaymentAccount = async (req, res) => {
...
+
+    await payment.save();
+
+    return "Your Action has been done successfully";
+  } catch (error) {
+    return "Server error";
+  }
+};
+
 exports.Payment = async (req, res) => {
```
Comment (OssamaLouati) [inline paiment-service/controllers/PaymentController.js]: weather removed or not after testing

**Review by OssamaLouati — COMMENTED** [review]
Finished review

**Review by BenomarAchraf — COMMENTED** [review]
Resolved all comments so far

**Pull request closed without merging**